numcodecs/blosc.pyx

```diff
@@ -486,6 +564,11 @@ class Blosc(Codec):
         buf = ensure_contiguous_ndarray(buf, self.max_buffer_size)
         return decompress(buf, out)
 
+    def decode_partial(self, buf, int start, int nitems, int typesize, int encoding_size, out=None):
+        buf = ensure_contiguous_ndarray(buf, self.max_buffer_size)
+        return decompress_partial(buf, start, nitems,
+                typesize, encoding_size, dest=out)
+
```

Comment:
Here we _could_ try to add an option to "guess" the encoding size, if we know the typesize then we know blosc will return _at least_ `typesize` bytes if we ask to decode the first element. 

IIRC when blosc_getitem succeed it tells you how much it wrote to the buffer you gave it, so if  you might be able to figure out whether the encoded typesize is correct, or 1 (if stored as bytes). 

Not sure it is worth it though and there is a performance impact to chec that as you need to decompress twice.

Reply:
We could also just peek at buf[3]... 
https://github.com/Blosc/c-blosc/blob/403d50ced2cd35d3e72a007d29d73110ac45e62b/blosc/blosc.c#L1613

Reply:
I added some code to infer the typesize / encoding size as you suggested here, and its passing the tests. I changed typesize and encoding size to kwargs that default to 0, and if they are zero, they use the inferences to do the partial decompression